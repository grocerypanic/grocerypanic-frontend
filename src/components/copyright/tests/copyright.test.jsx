import React from "react";
import { render, cleanup } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import Copyright from "../copyright.component";

import Strings from "../../../configuration/strings";
import Routes from "../../../configuration/routes";
import Assets from "../../../configuration/assets";

// Freeze Time
Date.now = jest.fn(() => new Date("2019-06-16T11:01:58.135Z"));

describe("Setup Environment", () => {
  let utils;
  const year = "2019";

  beforeEach(() => {
    utils = render(
      <MemoryRouter initialEntries={["/anything"]}>
        <Copyright />
      </MemoryRouter>
    );
  });

  afterEach(cleanup);

  it("should render with the correct message content", () => {
    const node = utils.getByTestId("CopyRight");
    expect(node).toBeTruthy();
    const [declaration, link, space, yearEnding] = Array.from(node.childNodes);
    expect(declaration.textContent).toBe(Strings.Copyight.CopyrightDeclaration);
    expect(link.textContent).toBe(`${Strings.Copyight.CopyrightMessage}`);
    expect(link.getAttribute("href")).toEqual(Routes.menu);
    expect(space.textContent).toBe(`${Assets.nonBreakingSpace}`);
    expect(yearEnding.textContent).toBe(`${year}.`);
  });

  it("should match the snapshot on file (styles)", () => {
    expect(utils.container.firstChild).toMatchSnapshot();
  });
});
