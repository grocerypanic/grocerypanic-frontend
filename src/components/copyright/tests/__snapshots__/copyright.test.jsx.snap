// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setup Environment should match the snapshot on file (styles) 1`] = `
.c0 {
  padding-top: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 0.7em;
  color: #000000;
}

.c1 {
  cursor: pointer;
  color: #006600;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div
  class="c0"
  data-testid="CopyRight"
>
  Created by
  <a
    class="c1"
    href="/"
  >
    Shared Vision Solutions
  </a>
  Â 
  2019.
</div>
`;
